### PR TITLE
add marketing

### PR DESCRIPTION
**Pull request by jawwadmukhtargmail**

### CODE REVIEW

File: UserService/UserService.Core/Services/CommunicationService.cs [hunk 1]
```diff
@@ -21,6 +21,38 @@ public CommunicationService(IHttpClientWrapper httpClientWrapper)
...
         }
 
+        public async Task<bool> DeleteMarketingContactAsync(DeleteMarketingContactRequest request, CancellationToken cancellationToken)
+        {
+            string path = $"api/DeleteMarketingContact";
+            var jsonContent = new StringContent(JsonConvert.SerializeObject(request), Encoding.UTF8, "application/json");
+            using (HttpResponseMessage response = await _httpClientWrapper.DeleteAsync(HttpClientConfigName.CommunicationService, path, jsonContent, cancellationToken).ConfigureAwait(false))
+            {
+                string jsonResponse = await response.Content.ReadAsStringAsync();
+                var emailSentResponse = JsonConvert.DeserializeObject<ResponseWrapper<bool, CommunicationServiceErrorCode>>(jsonResponse);
```
Comment (mike-f50) [inline UserService/UserService.Core/Services/CommunicationService.cs]: Variable name could probably be improved ;-)

File: UserService/UserService.Core/Services/CommunicationService.cs [hunk 2]
```diff
@@ -21,6 +21,38 @@ public CommunicationService(IHttpClientWrapper httpClientWrapper)
...
+        }
+
+        public async Task<bool> PutNewMarketingContactAsync(PutNewMarketingContactRequest request, CancellationToken cancellationToken)
+        {
+            string path = $"api/PutNewMarketingContact";
+            var jsonContent = new StringContent(JsonConvert.SerializeObject(request), Encoding.UTF8, "application/json");
+            using (HttpResponseMessage response = await _httpClientWrapper.PutAsync(HttpClientConfigName.CommunicationService, path, jsonContent, cancellationToken).ConfigureAwait(false))
+            {
+                string jsonResponse = await response.Content.ReadAsStringAsync();
+                var emailSentResponse = JsonConvert.DeserializeObject<ResponseWrapper<bool, CommunicationServiceErrorCode>>(jsonResponse);
```
Comment (mike-f50) [inline UserService/UserService.Core/Services/CommunicationService.cs]: And again

File: UserService/UserService.Handlers/DeleteUserHandler.cs [hunk 3]
```diff
@@ -38,6 +41,10 @@ public async Task<DeleteUserResponse> Handle(DeleteUserRequest request, Cancella
                     if(success)
                     {
                         success = await _repository.DeleteUserAsync(request.UserID, cancellationToken);
+                        await _communicationService.DeleteMarketingContactAsync(new DeleteMarketingContactRequest()
```
Comment (mike-f50) [inline UserService/UserService.Handlers/DeleteUserHandler.cs]: Throw an exception if it fails?

File: UserService/UserService.Handlers/PutModifyRegistrationPageTwoHandler.cs [hunk 4]
```diff
@@ -6,23 +6,44 @@
...
+        public async Task<PutModifyRegistrationPageTwoResponse> Handle(PutModifyRegistrationPageTwoRequest request, CancellationToken cancellationToken)
         {
             int response = _repository.ModifyUserRegistrationPageTwo(request.RegistrationStepTwo);
 
-            return Task.FromResult(new PutModifyRegistrationPageTwoResponse() { ID = response});
+            var user =_repository.GetUserByID(response);
+
+            if (user != null)
+            {
+                bool addedContact = await _communicationService.PutNewMarketingContactAsync(new PutNewMarketingContactRequest()
```
Comment (mike-f50) [inline UserService/UserService.Handlers/PutModifyRegistrationPageTwoHandler.cs:33]: Again, should probably do something with the response...  Ideally we'd log the error but not interrupt the flow in this scenario, but I'm not sure we have somewhere to log it currently, where we'd notice that it had happened.

**Review by mike-f50 — APPROVED** [review]

File: UserService/UserService.Handlers/DeleteUserHandler.cs (hunk 3 above)
Comment (jawwadmukhtargmail) [reply UserService/UserService.Handlers/DeleteUserHandler.cs]: done

**Pull request merged**